docs/resources/oxide_silo.md

```diff
@@ -0,0 +1,100 @@
+---
+page_title: "oxide_silo Resource - terraform-provider-oxide"
+---
+
+# oxide_silo (Resource)
+
+This resource manages the creation of an Oxide silo.
+
+-> This resource requires Terraform version v1.11 or later due to the
+`tls_certificates` attribute being a
+[write-only attribute](https://developer.hashicorp.com/terraform/plugin/framework/resources/write-only-arguments).
+
+-> Only the `quotas` attribute supports in-place modification. Changes to other
+attributes will result in the silo being destroyed and created anew.
+
+## Example Usage
+
+```hcl
+resource "oxide_silo" "example" {
+  name             = "showcase"
+  description      = "Demo and event silo."
+  admin_group_name = "showcase_admin"
+  identity_mode    = "saml_jit"
+  discoverable     = true
+  mapped_fleet_roles = {
+    admin  = ["admin", "collaborator"]
+    viewer = ["viewer"]
+  }
+  quotas = {
+    cpus    = 64
+    memory  = 137438953472 # 128 GiB
+    storage = 549755813888 # 512 GiB
+  }
+  tls_certificates = [
+    {
+      name        = "wildcard_cert"
+      description = "Wildcard cert for *.sys.oxide.example.com."
+      cert        = file("cert.pem")
+      key         = file("key.pem")
+      service     = "external_api"
+    },
+  ]
+}
+```
+
+## Schema
+
+### Required
+
+- `name` (String) Name of the Oxide silo.
+- `description` (String) Description for the Oxide silo.
+- `quotas` (Set of Object) Limits the amount of provisionable CPU, memory, and storage in the silo. (See [below for nested schema](#nestedatt--quotas).)
+- `tls_certificates` (String, Write-only) Initial TLS certificates to be used for the new silo's console and API endpoints. This attribute is a [write-only attribute](https://developer.hashicorp.com/terraform/plugin/framework/resources/write-only-arguments) and can only be modified by updating its configuration and [replacing the resource](https://developer.hashicorp.com/terraform/cli/state/taint). (See [below for nested schema](#nestedatt--tls).)
+- `discoverable` (Boolean) Whether this silo is present in the silo_list output. Defaults to `true`.
+
+### Optional
+
+- `identity_mode` (String) How identities are managed and users are authenticated in this silo. Valid values are `saml_jit` and `local_only`. Defaults to `local_only`.
+- `admin_group_name` (String) This group will be created during silo creation and granted the "Silo Admin" role. Identity providers can assert that users belong to this group and those users can log in and further initialize the Silo.
+- `mapped_fleet_roles` (Map) Mapping of which fleet roles are conferred by each silo role. The default is that no fleet roles are conferred by any silo roles unless there's a corresponding entry in this map. Map keys must be one of `admin`, `collaborator`, or `viewer` and map values are an array of strings that must be one of `admin`, `collaborator`, or `viewer`.
```

Comment:
This paragraph is a little confusing. Could you rephrase?

Reply:
I was a little confused on this myself when writing it. The API documentation is a bit lacking here so I tested this by hitting the API directly and reading the error messages that read that only `admin`, `collaborator`, and `viewer` can be valid keys and valid values in the array.

Reply:
Thanks! I'm guessing we're assuming the users know what they're doing here? Do we have a link to some docs explaining how this works? For example, what would something like `"viewer"={admin, collaborator}` mean?

Reply:
This was updated via another discussion on this pull request. Resolving.